# 🗺️ Master Lesson Plan: The Magic of Infinity and Irrationals
**Grade Level:** 6th Grade (Advanced/Enrichment)  
**Objective:** Students will explore how infinite steps can create finite numbers, discover patterns in repeating decimals, and learn how to prove a number is irrational using logic.

## 🐢 Section 1: Zeno's Paradox (Achilles and the Tortoise)

![Zeno's Paradox Time Steps](zenos_paradox.png)

Imagine a race between the legendary hero **Achilles** and a slow **Tortoise**. 
* Achilles runs at $10\text{ m/s}$.
* The Tortoise crawls at $1\text{ m/s}$.
* The Tortoise gets a $10\text{ meter}$ head start!

### The Paradox of Time Steps
Zeno argued that Achilles would never catch the Tortoise because every time Achilles reaches where the Tortoise *was*, the Tortoise has moved slightly ahead. Let's look at the **time intervals** (in seconds) it takes Achilles to perform each catch-up sprint:

1. **Step 1:** Achilles runs the first $10\text{ meters}$ to reach the Tortoise's starting point. This takes him **$1\text{ second}$** ($10\text{ m} / 10\text{ m/s}$). In that $1\text{ second}$, the Tortoise moves **$1\text{ meter}$** ($1\text{ m/s} \times 1\text{ s}$).
2. **Step 2:** Achilles runs that next $1\text{ meter}$ gap. This takes him **$0.1\text{ seconds}$** ($1\text{ m} / 10\text{ m/s}$). In that $0.1\text{ second}$, the Tortoise moves **$0.1\text{ meters}$**.
3. **Step 3:** Achilles runs that next $0.1\text{ meter}$ gap. This takes him **$0.01\text{ seconds}$** ($0.1\text{ m} / 10\text{ m/s}$). In that time, the Tortoise moves **$0.01\text{ meters}$**.

If we sum all of these infinite individual time intervals, we get an infinite series for total time ($T$):

$$T = 1 + 0.1 + 0.01 + 0.001 + 0.0001 + \dots$$

Let's use Python to see if this sum of infinite time segments traps Achilles in a loop forever, or if it converges cleanly to a finite moment!

In [1]:
# Simulating Achilles' Catch-up Time over 15 tracking cycles
terms = 15
current_time_interval = 1.0
total_time = 0.0

print(f"{'Step':<6} | {'Time Added (seconds)':<22} | {'Total Time Elapsed (s)':<22}")
print("-" * 58)

for step in range(1, terms + 1):
    total_time += current_time_interval
    print(f"{step:<6} | {current_time_interval:<22.15f} | {total_time:<22.15f}")
    current_time_interval *= 0.1  # Each time step shrinks to 1/10th of the previous

print("\nResult:")
print(f"Python time total after {terms} steps: {total_time:.15f} seconds")
print(f"Exact fractional value (10/9):       {10/9:.15f} seconds")

Step   | Time Added (seconds)   | Total Time Elapsed (s)
----------------------------------------------------------
1      | 1.000000000000000      | 1.000000000000000     
2      | 0.100000000000000      | 1.100000000000000     
3      | 0.010000000000000      | 1.110000000000000     
4      | 0.001000000000000      | 1.111000000000000     
5      | 0.000100000000000      | 1.111100000000000     
6      | 0.000010000000000      | 1.111110000000000     
7      | 0.000001000000000      | 1.111111000000000     
8      | 0.000000100000000      | 1.111111100000000     
9      | 0.000000010000000      | 1.111111110000000     
10     | 0.000000001000000      | 1.111111111000000     
11     | 0.000000000100000      | 1.111111111100000     
12     | 0.000000000010000      | 1.111111111110000     
13     | 0.000000000001000      | 1.111111111111000     
14     | 0.000000000000100      | 1.111111111111100     
15     | 0.000000000000010      | 1.111111111111110     

Result:
Python time total af

## 🔢 Section 2: Proving the Sum Using Algebraic Substitution

Python shows us that the total time approaches $1.1111\dots$ seconds, which looks like the fraction $\frac{10}{9}$. Let's prove it perfectly using algebra!

### Step 1: Write the equation
Let $T$ equal our infinite sum of time steps:
$$T = 1 + 0.1 + 0.01 + 0.001 + \dots$$

### Step 2: Multiply by 10
Multiply both sides by $10$ to shift the decimal point one place to the right:
$$10T = 10 + 1 + 0.1 + 0.01 + 0.001 + \dots$$

### Step 3: Substitute $T$ back in
Notice that the entire decimal tail is identical to our original definition of $T$:
$$10T = 10 + \underbrace{(1 + 0.1 + 0.01 + \dots)}_{T}$$
$$10T = 10 + T$$

### Step 4: Solve for $T$
Subtract $T$ from both sides:
$$9T = 10$$

Divide by $9$:
$$T = \frac{10}{9}$$

**Conclusion:** Achilles catches up and overtakes the Tortoise at exactly $\frac{10}{9}$ seconds (or $1.111\dots$ seconds)!

## 💡 Section 3: Exploring More Repeating Decimals

Let's test this exact same substitution trick on other repeating decimal properties.

### Exploration A: Proof that $0.999\dots = 1$
1. Let $x = 0.9999\dots$
2. Multiply by $10$: $10x = 9.9999\dots$
3. Separate the integer: $10x = 9 + 0.9999\dots$
4. Substitute $x$: $10x = 9 + x$
5. Subtract $x$: $9x = 9$
6. Divide by $9$: $x = 1$

### Exploration B: Proof that $0.121212\dots = \frac{4}{33}$
Because the pattern repeats every **two** digits, we multiply by $100$ instead:
1. Let $y = 0.121212\dots$
2. Multiply by $100$: $100y = 12.121212\dots$
3. Substitute $y$: $100y = 12 + y$
4. Subtract $y$: $99y = 12$
5. Divide by $99$: $y = \frac{12}{99}$ 
6. Simplify by dividing the top and bottom by $3$: $y = \frac{4}{33}$

## ⚡ Section 4: Shortcuts and Mixed Decimals

### The Pure Repeating Shortcut
If a repeating pattern starts right after the decimal, you can build a fraction instantly:
* **Numerator:** The repeating digits.
* **Denominator:** A $9$ for every digit in the repeating block.
* *Example:* $0.777\dots = \frac{7}{9}$
* *Example:* $0.315315\dots = \frac{315}{999} = \frac{35}{111}$

### What if it's a Mixed Repeating Decimal? ($0.18333\dots$)
When a decimal has a non-repeating section before the loop starts, we can use two different multipliers to align and subtract the tails. Let's see how Python calculates this exact value:

In [ ]:
# Testing the Mixed Fraction 11/60
fraction_value = 11 / 60
print(f"The fraction 11/60 evaluates to: {fraction_value}")

### Algebraic Breakdown for Mixed Decimals:
1. Let $z = 0.18333\dots$
2. Multiply by $100$ to get to the start of the repeating block: 
   $$100z = 18.333\dots \quad \text{(Equation A)}$$
3. Multiply original $z$ by $1000$ to go past one full repeat cycle: 
   $$1000z = 183.333\dots \quad \text{(Equation B)}$$
4. Subtract Equation A from Equation B to erase the infinite tail of $3$s:
   $$1000z - 100z = 183.333\dots - 18.333\dots$$
   $$900z = 165$$
5. Solve and simplify: 
   $$z = \frac{165}{900} = \frac{11}{60}$$

## 🚫 Section 5: When the Trick Fails (Irrational Numbers)

All the numbers we looked at so far are **Rational Numbers** because they can turn into a ratio (a fraction $\frac{p}{q}$).

An **Irrational Number** is a number whose decimal expansion goes on forever **without ever repeating a pattern**.

### Why the trick fails:
Think about $\pi = 3.14159265\dots$
If you multiply by $10$, you get $31.4159265\dots$. If you try to subtract them, the decimals will not align or cancel out because the sequences are completely random and non-repeating. You can never get rid of the infinite trail!

## 🧠 Section 6: Proof by Contradiction (The Mystery of $\sqrt{2}$)

How do we prove a number is completely impossible to write as a fraction? We use a powerful logical tool called **Proof by Contradiction**. We assume the opposite is true, and show it creates an impossible rule break!

### 1. The Setup (The Assumption)
Assume $\sqrt{2}$ **is** a rational number. That means it can be written as a fraction:
$$\sqrt{2} = \frac{a}{b}$$
*Crucial Rule:* This fraction $\frac{a}{b}$ is **fully simplified**. This means $a$ and $b$ share no common factors. They cannot both be even numbers!

### 2. Squaring and Rearranging
Square both sides to remove the root:
$$2 = \frac{a^2}{b^2}$$

Multiply by $b^2$:
$$2b^2 = a^2$$

### 3. Finding the Secret Factor
Look at the equation $a^2 = 2b^2$. Because $a^2$ equals $2$ times an integer, **$a^2$ must be an even number**. 
* Rule: If a square number is even, the original root must be even too.
* Therefore, **$a$ is an even number**.

Since $a$ is even, we can write it as $2k$ (where $k$ is another integer). Let's swap it back into our equation:
$$2b^2 = (2k)^2$$
$$2b^2 = 4k^2$$

Divide both sides by $2$:
$$b^2 = 2k^2$$

### 4. The Second Secret Factor
Look at our new equation $b^2 = 2k^2$. By the exact same logic, $b^2$ is even, which means **$b$ must be an even number** too!

### 💥 The Core Contradiction!
We have mathematically proven that **both $a$ and $b$ are even numbers**. 
But wait! If they are both even, the fraction $\frac{a}{b}$ can be simplified by dividing by $2$. This completely violates our very first setup rule that the fraction was fully simplified!

### Final Conclusion
Our logic was perfect, so our starting assumption must be wrong. It is impossible for $\sqrt{2}$ to be written as a fraction. $\sqrt{2}$ is Irrational!

## 📝 Section 7: Comprehension Check Questions

Answer the following questions based on the lessons and notebook executions above.

### Part 1: Zeno's Paradox & The Algebraic Trick
1. Conceptual: If Zeno's paradox says Achilles must complete an infinite number of steps to reach the Tortoise, why does he not run forever? Explain using the concept of a series that converges.
2. Algebraic: In Section 2, why did we multiply the equation by 10 instead of another number? What did multiplying by 10 achieve?
3. Application: Using the shortcut rule from Section 4, write the repeating decimal $0.555\dots$ as a fraction.

### Part 2: Irrationals & Proof by Contradiction
4. Logical: In the proof that $\sqrt{2}$ is irrational, what was our starting assumption?
5. Critical Thinking: Explain the specific rule break (the contradiction) that made our starting assumption false.
6. Comparison: Why can we use algebraic substitution to turn $0.121212\dots$ into a fraction, but we cannot use it to turn $\pi$ ($3.14159\dots$) into a fraction?